src/main/java/com/wp/library/Book/domain/adapter/BookAdapter.java

```diff
@@ -0,0 +1,9 @@
+package com.wp.library.Book.domain.adapter;
+
+import com.wp.library.Book.domain.contract.BookRequest;
+import com.wp.library.Book.domain.contract.BookResponse;
+
+public interface BookAdapter {
+
+    BookResponse createBook(BookRequest request);
```

Comment:
You could add here also createEbook and createPrintedBook. Then in BookController you could use BookAdapter instead of BookService.

Reply:
Yes, I thought about it, but I wanted to add a method called factory to better show that it is a factory pattern xD